PG 18 branch, pglast compilation failure on strlcpy() in some GH/CI combo

As previously reported in [this comment](https://github.com/pganalyze/libpg_query/issues/304#issuecomment-3696031130), I'm still [seeing](https://github.com/lelit/pglast/actions/runs/26330568990/job/77515862475#step:5:573) 
```
  src/postgres/src_backend_utils_cache_syscache.c:182:5: error: implicit declaration of function ‘strlcpy’; did you mean ‘strncpy’? [-Wimplicit-function-declaration]
    182 |     strlcpy(NameStr(t->typname), bt->typname, NAMEDATALEN);
        |     ^~~~~~~
        |     strncpy
  make: *** [Makefile:185: src/postgres/src_backend_utils_cache_syscache.o] Error 1
```

The strange thing is that other combos ([the main one](https://github.com/lelit/pglast/actions/runs/26330568990/job/77515862456) for example) do not exhibit the problem.

--- a/scripts/pg_config_overrides.h
+++ b/scripts/pg_config_overrides.h
@@ -49,6 +49,21 @@
 #define HAVE_DECL_STRCHRNUL 0
 #endif
 
+/*
+ * Ensure we use built-in strlcpy/strlcat on systems that have them (all BSDs, macOS, musl, and glibc 2.38+).
+ *
+ * See https://sourceware.org/git/?p=glibc.git;a=commit;h=454a20c8756c9c1d55419153255fc7692b3d2199 re: glibc.
+ */
+#undef HAVE_DECL_STRLCPY
+#undef HAVE_DECL_STRLCAT
+#if !defined(_WIN32) && !defined(_WIN64) && (defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__APPLE__) || !defined(__GLIBC__) || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 38) || __GLIBC__ > 2)
+#define HAVE_DECL_STRLCPY 1
+#define HAVE_DECL_STRLCAT 1
+#else
+#define HAVE_DECL_STRLCPY 0
+#define HAVE_DECL_STRLCAT 0
+#endif
+
 /* 32-bit */
 #if defined(_WIN32) || __SIZEOF_POINTER__ == 4
 #undef ALIGNOF_DOUBLE
@@ -82,10 +97,6 @@
 #define HAVE_DECL_PREADV 0
 #undef HAVE_DECL_PWRITEV
 #define HAVE_DECL_PWRITEV 0
-#undef HAVE_DECL_STRLCAT
-#define HAVE_DECL_STRLCAT 0
-#undef HAVE_DECL_STRLCPY
-#define HAVE_DECL_STRLCPY 0
 #undef HAVE_GETIFADDRS
 #undef HAVE_GETPEEREID
 #undef HAVE_IFADDRS_H

--- a/src/postgres/include/pg_config.h
+++ b/src/postgres/include/pg_config.h
@@ -873,6 +873,21 @@
 #define HAVE_DECL_STRCHRNUL 0
 #endif
 
+/*
+ * Ensure we use built-in strlcpy/strlcat on systems that have them (all BSDs, macOS, musl, and glibc 2.38+).
+ *
+ * See https://sourceware.org/git/?p=glibc.git;a=commit;h=454a20c8756c9c1d55419153255fc7692b3d2199 re: glibc.
+ */
+#undef HAVE_DECL_STRLCPY
+#undef HAVE_DECL_STRLCAT
+#if !defined(_WIN32) && !defined(_WIN64) && (defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__APPLE__) || !defined(__GLIBC__) || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 38) || __GLIBC__ > 2)
+#define HAVE_DECL_STRLCPY 1
+#define HAVE_DECL_STRLCAT 1
+#else
+#define HAVE_DECL_STRLCPY 0
+#define HAVE_DECL_STRLCAT 0
+#endif
+
 /* 32-bit */
 #if defined(_WIN32) || __SIZEOF_POINTER__ == 4
 #undef ALIGNOF_DOUBLE
@@ -906,10 +921,6 @@
 #define HAVE_DECL_PREADV 0
 #undef HAVE_DECL_PWRITEV
 #define HAVE_DECL_PWRITEV 0
-#undef HAVE_DECL_STRLCAT
-#define HAVE_DECL_STRLCAT 0
-#undef HAVE_DECL_STRLCPY
-#define HAVE_DECL_STRLCPY 0
 #undef HAVE_GETIFADDRS
 #undef HAVE_GETPEEREID
 #undef HAVE_IFADDRS_H

